apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/UnclaimedActivityRecordIntegrationTest.scala

```diff
@@ -0,0 +1,311 @@
+import org.lfdecentralizedtrust.splice.codegen.java.splice.dsorules.UnallocatedUnclaimedActivityRecord
+import org.lfdecentralizedtrust.splice.integration.EnvironmentDefinition
+import org.lfdecentralizedtrust.splice.sv.automation.delegatebased.{
+  AllocateUnallocatedUnclaimedActivityRecordTrigger,
+  ExpiredUnallocatedUnclaimedActivityRecordTrigger,
+  ExpiredUnclaimedActivityRecordTrigger,
+}
+import org.lfdecentralizedtrust.splice.util.TriggerTestUtil
+import org.lfdecentralizedtrust.splice.util.WalletTestUtil
+import org.lfdecentralizedtrust.splice.wallet.automation.CollectRewardsAndMergeAmuletsTrigger
+
+import com.daml.ledger.javaapi.data.Identifier
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class UnclaimedActivityRecordIntegrationTest
+    extends SvIntegrationTestBase
+    with TriggerTestUtil
+    with WalletTestUtil {
+
+  override def environmentDefinition
+      : org.lfdecentralizedtrust.splice.integration.EnvironmentDefinition =
+    EnvironmentDefinition
+      .simpleTopology1Sv(this.getClass.getSimpleName)
+
+  override protected lazy val sanityChecksIgnoredRootCreates: Seq[Identifier] = Seq(
+    UnclaimedReward.TEMPLATE_ID_WITH_PACKAGE_ID
+  )
+
+  "An UnallocatedUnclaimedActivityRecord gets expired" in { implicit env =>
+    val sv1Party = sv1Backend.getDsoInfo().svParty
+    val aliceUserName = aliceWalletClient.config.ledgerApiUser
+    val aliceParty = aliceValidatorBackend.onboardUser(aliceUserName)
+    val amountToMint = 15.0
+
+    def allocateTriggers: Seq[Trigger] =
+     activeSvs.map(_.dsoDelegateBasedAutomation.trigger[AllocateUnallocatedUnclaimedActivityRecordTrigger])
+    def expiredUnallocatedTriggers: Seq[Trigger] =
+     activeSvs.map(_.dsoDelegateBasedAutomation.trigger[ExpiredUnallocatedUnclaimedActivityRecordTrigger])
+    setTriggersWithin(
+     triggersToPauseAtStart = allocateTriggers ++ expiredUnallocatedTriggers,
+     triggersToResumeAtStart = Seq.empty,
+    ) {
+     actAndCheck(
+       "Creating vote request", {
+         val action = new ARC_DsoRules(
+           new SRARC_CreateUnallocatedUnclaimedActivityRecord(
+             new DsoRules_CreateUnallocatedUnclaimedActivityRecord(
+               aliceParty.toProtoPrimitive,
+               BigDecimal(amountToMint).bigDecimal,
+               "alice is doing great - vote",
+               Instant.now().plus(5, ChronoUnit.SECONDS),
+             )
+           )
+         )
+         eventuallySucceeds() {
+           sv1Backend.createVoteRequest(
+             sv1Party.toProtoPrimitive,
+             action,
+             "url",
+             "alice is doing great",
+             sv1Backend.getDsoInfo().dsoRules.payload.config.voteRequestTimeout,
+             None,
+           )
+         }
+       },
+     )("vote request and UnallocatedUnclaimedActivityRecord have been created",
+        _ => {
+         sv1Backend.listVoteRequests().loneElement
+          eventually(){
```

Comment:
actAndCheck automatically wraps the second `check` part into an `eventually` so this is redundant